js/src/jsonrpc/index.spec.js

```diff
@@ -14,37 +14,113 @@
 // You should have received a copy of the GNU General Public License
 // along with Parity.  If not, see <http://www.gnu.org/licenses/>.
 
+import fs from 'fs';
+import path from 'path';
 import interfaces from './';
-import { Address, BlockNumber, Data, Hash, Integer, Quantity } from './types';
+import * as customTypes from './types';
 
-const flatlist = {};
+const allowedTypes = [Array, Boolean, Object, String];
+
+Object.keys(customTypes).map((key) => {
+  allowedTypes.push(customTypes[key]);
+});
 
 function verifyType (obj) {
   if (typeof obj !== 'string') {
-    expect(obj).to.satisfy(() => {
-      return obj.type === Array ||
-        obj.type === Boolean ||
-        obj.type === Object ||
-        obj.type === String ||
-        obj.type === Address ||
-        obj.type === BlockNumber ||
-        obj.type === Data ||
-        obj.type === Hash ||
-        obj.type === Integer ||
-        obj.type === Quantity;
-    });
+    expect(obj).to.satisfy(() => allowedTypes.indexOf(obj.type) >= 0);
   }
 }
 
+// Get a list of JSON-RPC from Rust trait source code
+function parseMethodsFromRust (source) {
+  // Matching the custom `rpc` attribute with it's doc comment
+  const attributePattern = /((?:\s*\/\/\/.*$)*)\s*#\[rpc\(([^)]+)\)]/gm;
+  const commentPattern = /\s*\/\/\/\s*/g;
+  const separatorPattern = /\s*,\s*/g;
+  const assignPattern = /([\S]+)\s*=\s*"([^"]*)"/;
+  const ignorePattern = /@(deprecated|unimplemented|alias)\b/i;
+
+  const methods = [];
+
+  source.toString().replace(attributePattern, (match, comment, props) => {
+    comment = comment.replace(commentPattern, '\n').trim();
+
+    // Skip deprecated methods
+    if (ignorePattern.test(comment)) {
+      return match;
+    }
+
+    props.split(separatorPattern).forEach((prop) => {
+      const [, key, value] = prop.split(assignPattern) || [];
+
+      if (key === 'name' && value != null) {
+        methods.push(value);
+      }
+    });
+
+    return match;
+  });
+
+  return methods;
+}
+
+// Get a list of all JSON-RPC methods from all defined traits
+function getMethodsFromRustTraits () {
+  const traitsDir = path.join(__dirname, '../../../rpc/src/v1/traits');
+
+  return [
+    'eth',
+    'eth_signing', // Rolled into `eth`
+    'net',
+    'parity',
+    'parity_accounts',
+    'parity_set',
+    'parity_signing', // Rolled into `parity`
+    'personal',
+    'signer',
+    'traces',
+    'web3'
+  ].map((name) => fs.readFileSync(path.join(traitsDir, `${name}.rs`)))
+    .map(parseMethodsFromRust)
+    .reduce((a, b) => a.concat(b));
+}
+
+const rustMethods = {};
+
+getMethodsFromRustTraits().sort().forEach((method) => {
+  const [group, name] = method.split('_');
+
+  rustMethods[group] = rustMethods[group] || {};
+  rustMethods[group][name] = true;
+});
+
+describe('Rust defined JSON-RPC methods', () => {
```

Comment:
Might want to follow the _test names convention_ with `jsonrpc/interfaces` as the top test.